frontend/lib/item_listing/view_models/listing.viewmodel.dart

```diff
@@ -1,37 +1,106 @@
-import 'package:flutter/foundation.dart';
-
+import 'package:flutter/material.dart';
+import 'package:utm_marketplace/item_listing/model/filter_options.model.dart';
 import 'package:utm_marketplace/item_listing/model/listing.model.dart';
 import 'package:utm_marketplace/item_listing/repository/listing_repo.dart';
-
 import 'package:utm_marketplace/shared/view_models/loading.viewmodel.dart';
+import 'package:utm_marketplace/item_listing/components/filter_bottom_sheet/filter_bottom_sheet.component.dart';
 
 class ListingViewModel extends LoadingViewModel {
   ListingViewModel({
     required this.repo,
   });
 
   final ListingRepo repo;
+  List<Item> _allItems = [];
+  List<Item> _filteredItems = [];
 
-  ListingModel _listingModel = ListingModel();
-
-  List<Item> _items = [];
-  List<Item> get items => _items;
-  set items(List<Item> value) {
-    _items = value;
-    notifyListeners();
-  }
+  List<Item> get items => _filteredItems;
 
   Future<void> fetchData() async {
     try {
       isLoading = true;
-
-      _listingModel = await repo.fetchData();
-      items = _listingModel.items;
+      final model = await repo.fetchData();
+      _allItems = model.items;
+      _filteredItems = _allItems;
+      notifyListeners();
     } catch (exc) {
       debugPrint('Error in fetchData : ${exc.toString()}');
     } finally {
       isLoading = false;
       notifyListeners();
     }
   }
+
+  void applyFilters(FilterOptions filters) {
+    _filteredItems = List.from(_allItems);
+
```

Comment:
Sorry but I think we can refactor this to be a single for loop checking each condition for each item. This implementation would have you go through the list multiple times. :,)

Reply:
ordering should still be done after like you did, though